src/SimpleGraphs/simplegraph.jl

```diff
@@ -21,6 +21,26 @@ end
 
 eltype(x::SimpleGraph{T}) where {T} = T
 
+# SimpleGraph(g)
+"""
+    SimpleGraph{T}(g::AbstractGraph)
+
+Construct a `SimpleGraph{T}` from an `AbstractGraph`. 
+If `g` is a `SimpleDiGraph`, then every directed edge in `g` is 
+added as an undirected edge.
+"""
+SimpleGraph(g::AbstractGraph) = SimpleGraph{Int}(g)
+
+# SimpleGraph{UInt8}(g)
+function SimpleGraph{T}(g::AbstractGraph) where {T<:Integer}
+    simpleGraph = SimpleGraph(nv(g))
+    @inbounds for e in edges(g)
```

Comment:
I don't think that `@inbounds` has any effect here - the only place were some bound checks can happen in this loop are inside the `add_edge!` function, but that function does not allow the propagation of `@inbounds` anyways.